src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -64,26 +62,15 @@ public DateTime getTimestamp()
   @Override
   public List<String> getDimension(String dimension)
   {
-    Object dimValue = event.get(dimension);
+    final Object dimValue = event.get(dimension);
 
     if (dimValue == null) {
-      return Lists.newArrayList();
+      return Collections.emptyList();
     } else if (dimValue instanceof List) {
-      return Lists.transform(
-          (List) dimValue,
-          new Function<Object, String>()
-          {
-            @Override
-            public String apply(Object input)
-            {
-              return String.valueOf(input);
-            }
-          }
-      );
-    } else if (dimValue instanceof Object) {
-      return Arrays.asList(String.valueOf(event.get(dimension)));
+      // guava's toString function fails on null objects, so please do not use it
+      return Lists.transform((List<Object>) dimValue, TO_STRING_INCLUDING_NULL);
     } else {
-      throw new IAE("Unknown dim type[%s]", dimValue.getClass());
```

Comment:
Can you explain why we are changing the behavior here? Maybe you can explain the problem you are addressing in your PR description?